x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ExceptionListClient } from '../../../../../lists/server';
+import { ENDPOINT_TRUSTED_APPS_LIST_ID } from '../../../../../lists/common/constants';
+
+import {
+  DeleteTrustedAppsRequestParams,
+  GetTrustedAppsListRequest,
+  GetTrustedListAppsResponse,
+  PostTrustedAppCreateRequest,
+  PostTrustedAppCreateResponse,
+} from '../../../../common/endpoint/types';
+
+import { exceptionItemToTrustedAppItem, newTrustedAppItemToExceptionItem } from './utils';
+
+export const deleteTrustedApp = async (
+  exceptionsListClient: ExceptionListClient,
+  { id }: DeleteTrustedAppsRequestParams
+) => {
+  await exceptionsListClient.deleteExceptionListItem({
+    id,
+    itemId: undefined,
+    namespaceType: 'agnostic',
+  });
+};
+
+export const getTrustedAppsList = async (
+  exceptionsListClient: ExceptionListClient,
+  { page, per_page: perPage }: GetTrustedAppsListRequest
+): Promise<GetTrustedListAppsResponse> => {
+  // Ensure list is created if it does not exist
+  await exceptionsListClient.createTrustedAppsList();
+
+  const results = await exceptionsListClient.findExceptionListItem({
+    listId: ENDPOINT_TRUSTED_APPS_LIST_ID,
+    page,
+    perPage,
+    filter: undefined,
+    namespaceType: 'agnostic',
+    sortField: 'name',
+    sortOrder: 'asc',
+  });
+
+  return {
+    data: results?.data.map(exceptionItemToTrustedAppItem) ?? [],
+    total: results?.total ?? 0,
+    page: results?.page ?? 1,
+    per_page: results?.per_page ?? perPage!,
```

Comment:
if `results.per_page` is not defined/null and it falls to `perPage` -- which is coming from the HTTP request. It looks like that is optional? I didn't see a default value provided in the call chain here, so what happens if that is also undefined, and the `!` fails?

Reply:
I personally would just make those arguments required in the schema, but this code was already in place, I just moved it around a bit.